experimental/list_formatter/src/string_matcher.rs

```diff
@@ -0,0 +1,40 @@
+#[cfg(any(test, feature = "provider_transform_internals"))]
+use crate::error::Error;
+use alloc::borrow::Cow;
+use icu_provider::yoke::{self, *};
+use regex::Regex;
+
+#[derive(Clone, Debug, PartialEq, Yokeable, ZeroCopyFrom)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Deserialize, serde::Serialize)
+)]
+pub(crate) struct StringMatcher<'data>(
+    #[cfg_attr(feature = "provider_serde", serde(borrow))] Cow<'data, str>,
+);
+
+impl<'data> StringMatcher<'data> {
+    #[cfg(any(test, feature = "provider_transform_internals"))]
+    pub(crate) fn new(pattern: &str) -> Result<Self, Error> {
+        let regex = alloc::format!("(?i)^({})", pattern);
+        Regex::new(&regex).map_err(Error::IllegalCondition)?;
+        Ok(StringMatcher(Cow::Owned(regex)))
+    }
+
+    pub(crate) fn test(&self, string: &str) -> bool {
+        Regex::new(&self.0).unwrap().is_match(string)
```

Comment:
Question: Do you still intend to use regex-automata? Do you have data on code size / performance / memory usage impact?

Reply:
I am, see https://github.com/robertbastian/icu4x/pull/2/files. How would I obtain such data?